File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/applikasjon/domenemodeller/ImporttidspunktDto.java
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.domenemodeller;

import java.sql.Timestamp;
import java.time.LocalDate;
import lombok.EqualsAndHashCode;
import lombok.Getter;

@Getter
@EqualsAndHashCode
public class ImporttidspunktDto {

  private final Timestamp sistImportertTidspunkt;
  private final ÅrstallOgKvartal gjeldendePeriode;

  public ImporttidspunktDto(Timestamp sistImportertTidspunkt, ÅrstallOgKvartal gjeldendePeriode) {
    this.sistImportertTidspunkt = sistImportertTidspunkt;
    this.gjeldendePeriode = gjeldendePeriode;
  }

  public LocalDate getImportertDato() {
    return sistImportertTidspunkt.toLocalDateTime().toLocalDate();
  }
}

File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/applikasjon/domenemodeller/ImporttidspunktDto.kt
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.domenemodeller

import java.sql.Timestamp
import java.time.LocalDate

data class ImporttidspunktDto(
    val sistImportertTidspunkt: Timestamp,
    val gjeldendePeriode: ÅrstallOgKvartal
) {
    val importertDato: LocalDate
        get() = sistImportertTidspunkt.toLocalDateTime().toLocalDate()
}
